src/overlays/actors/ovl_En_Zo/z_en_zo.c

```diff
@@ -78,41 +95,255 @@ static DamageTable D_8099F4E8 = {
+        Math_SmoothStepToS(&this->upperBodyRot.x, 0, 4, 1000, 1);
+        Math_SmoothStepToS(&this->upperBodyRot.y, 0, 4, 1000, 1);
+    }
+    EnZo_Blink(this, 3);
+    func_8013D9C8(globalCtx, this->limbRotY, this->limbRotZ, 20);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/func_8099E96C.s")
+void EnZo_Walk(EnZo* this, GlobalContext* globalCtx) {
+    if (ENZO_GET_PATH(&this->actor) != 0x3F) {
+        EnZo_SetAnimation(&this->skelAnime, 6);
+    }
+    if (ENZO_GET_PATH(&this->actor) != 0x3F) {
+        this->actionFunc = EnZo_FollowPath;
+    } else {
+        this->actionFunc = EnZo_DoNothing;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/func_8099E9E0.s")
+void EnZo_FollowPath(EnZo* this, GlobalContext* globalCtx) {
+    s16 speed;
+    Vec3f pos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/func_8099EA60.s")
+    Math_SmoothStepToF(&this->actor.speedXZ, 1.0f, 0.4f, 1000.0f, 0.0f);
+    speed = this->actor.speedXZ * 400.0f;
+    if (func_8013D68C(this->path, this->waypoint, &pos) && func_8013D768(&this->actor, &pos, speed)) {
+        this->waypoint++;
+        if (this->waypoint >= this->path->count) {
+            this->waypoint = 0;
+        }
+    }
+    if (this->actor.depthInWater > 60.0f) {
+        EnZo_SetAnimation(&this->skelAnime, 1);
+        this->actionFunc = EnZo_TreadWater;
+        this->actor.gravity = 0.0f;
+        this->actor.speedXZ = 0.0f;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/func_8099EBD8.s")
+void EnZo_TreadWater(EnZo* this, GlobalContext* globalCtx) {
+    f32 targetYVel;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/func_8099EC50.s")
+    if (this->actor.depthInWater < (sREG(0) + 50.0f)) {
+        targetYVel = -1.0f;
+    } else {
+        targetYVel = 1.0f;
+    }
+    Math_ApproachF(&this->actor.velocity.y, targetYVel, (sREG(1) + 18.0f) * 0.01f, (sREG(2) + 12.0f) * 0.01f);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/func_8099ED4C.s")
+void EnZo_DoNothing(EnZo* this, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/func_8099EE24.s")
+void EnZo_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnZo* this = THIS;
+    SkelAnime* skelAnime = &this->skelAnime;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Zo/EnZo_Init.s")
+    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 0.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_0600D208, NULL, this->jointTable, this->morphTable, 20);
+    EnZo_SetAnimation(skelAnime, 0);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+    this->path = func_8013D648(globalCtx, ENZO_GET_PATH(&this->actor), 0x3F);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actionFunc = EnZo_Walk;
+    this->actor.gravity = -4.0f;
```

Comment:
This is quite bunched up, I'd add some blank lines between the parts that are doing different things:
```suggestion
    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 0.0f);
    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_0600D208, NULL, this->jointTable, this->morphTable, 20);
    EnZo_SetAnimation(skelAnime, 0);

    Collider_InitCylinder(globalCtx, &this->collider);
    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
    CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);

    this->path = func_8013D648(globalCtx, ENZO_GET_PATH(&this->actor), 0x3F);
    Actor_SetScale(&this->actor, 0.01f);

    this->actionFunc = EnZo_Walk;
    this->actor.gravity = -4.0f;
```